src/app/cli/src/app.rs

```diff
@@ -87,13 +99,16 @@ pub async fn run(workspace_layout: WorkspaceLayout, args: cli::Cli) -> Result<()
 
     prepare_run_dir(&workspace_layout.run_info_dir);
 
-    let maybe_db_connection_settings = config
-        .database
-        .clone()
+    let is_init_command = maybe_init_multi_tenant_flag.is_some();
+    let app_database_config =
+        get_app_database_config(&config, is_multi_tenant_workspace, is_init_command);
+    let (database_config, maybe_sqlite_db_temp_dir) = app_database_config.into_inner();
```

Comment:
Minor: mentining SQlite at this particular place looks like too low-level detail. Can we hide it better?

Reply:
Agree. Did some refactoring to partially hide the extra context sticking out

Done